### PR TITLE
Don't split paragraphs in blockquotes

### LINKED ISSUE #139: Paragraphs in Blockquotes Split Text
Copying this issue from aaronsw/html2text#17, since it's still a problem.

If you have this construction
```
<blockquote>
    <p>Paragraph 1</p>
    <p>Paragraph 2</p>
</blockquote>
```
html2text generates this output
```
> Paragraph 1

>

> Paragraph 2
```
This should not be, since the empty line between the paragraphs breaks the blockquote...

### PR DESCRIPTION
**Pull request by jonathan-s**
This PR should fix #139. When fixing this issue I discovered that whitespace plays a crucial role at the end of the line which is rather annoying. This PR also partially fixes this as it removes whitespace from some lines.

### CODE REVIEW

**Comment by jonathan-s** [comment]
Ping @Alir3z4 :)

**Comment by Alir3z4** [comment]
@ciprianmiclaus Would be great to have you review it. Could you please have a look at this ?

**Comment by Alir3z4** [comment]
@jonathan-s the extra space at the end of each line in Markdown is rendered a breakline `<br>`, that br should not be removed.

If you look at some test files, for instance https://github.com/jonathan-s/html2text/blob/8cdc25471f1472d7cbb02f66658ea50419d68c05/test/normal.html#L72-L73 you'll notice it has `<br />` between both text, that should be translated to double Space `  ` in markdown.

I think most of the md file that have been modified in the pull request have been wrongly changed regarding their ending double spaces.

File: test/normal.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ text to separate lists
     2. a **bold human** being
   3. end
 
-**bold**
```
Comment (Alir3z4) [inline test/normal.md]: This is a `<br />` in Markdown.

File: test/normal.md [hunk 2]
```diff
@@ -16,8 +16,8 @@ text to separate lists
     2. a **bold human** being
   3. end
 
-**bold**  
-_italic_
```
Comment (Alir3z4) [inline test/normal.md]: This is a `<br />` in Markdown as well.

**Comment by jonathan-s** [comment]
With regards to breaklines being ignored. From my understanding the logic of breaklines are the following. At line [348](https://github.com/Alir3z4/html2text/blob/master/html2text/__init__.py#L348-L352) two whitespaces and a new line is introduced. When all markdown has been generated we make a final pass with `optwrap` on the text to handle paragraphs correctly. At line [922](https://github.com/Alir3z4/html2text/pull/179/files#diff-5e3a9e55d540708ee71bbae0911b45bbL922) if a line ends with two whitespaces only a single linebreak is added, so the linebreak from line 348 is not ignored. Instead of adding back the two whitespaces on line 922 I simply removed them. To the best of my knowledge two whitespaces have no inherent meaning in markdown, and `<br>` is preserved by adding a single linebreak.

**Comment by jonathan-s** [comment]
See the spec at [commonmark](http://spec.commonmark.org/0.28/#example-189)

**Comment by jonathan-s** [comment]
Just read the initial spec at daring fireball. There two trailing spaces has meaning. But I wonder how actual markdown parsers generate the html. What are some python libraries that take markdown and generates html?

**Comment by Alir3z4** [comment]
specs of commonmark also states 2 spaces for hard breaks: http://spec.commonmark.org/0.28/#hard-line-break

**Comment by Alir3z4** [comment]
Python Markdown module also converts 2 spaces to hard breakline. https://pypi.python.org/pypi/Markdown

Github Flavored Markdown also does that: 
For instance: 
```
line 1  
line 2  
```
 in Github flavored markdown would be 

line 1  
line 2

**Comment by jonathan-s** [comment]
I stand corrected. Well let me look this over and see if I can do something about it. Would you mind if I used `\` instead of two spaces like the alternative specifies? It's a lot more legible than spaces. 

```
foo\
baz
```

**Comment by Alir3z4** [comment]
> Would you mind if I used `\` instead of two spaces like the alternative specifies? It's a lot more legible than spaces.

Using `\` would not make the converted text eye candy, we want to keep the result similar to plain text that happens to be Markdown as well, right ? Using `\` would make it harsh. So let's stay with double spaces for hard break lines to keep the result clean.

**Comment by Alir3z4** [comment]
@jonathan-s any update on this pull request ?

**Comment by Alir3z4** [comment]
Closing due to being stale

**Pull request closed without merging**